vmprof/__init__.py

```diff
@@ -36,14 +43,41 @@ def enable(fileno, period=DEFAULT_PERIOD, memory=False, warn=True):
         if warn and sys.pypy_version_info[:3] < (4, 1, 0):
             print ("PyPy <4.1 have various kinds of bugs, pass warn=False if you know what you're doing")
             raise Exception("PyPy <4.1 have various kinds of bugs, pass warn=False if you know what you're doing")
-        _vmprof.enable(fileno, period)
+        gz_fileno = _gzip(fileno)
+        _vmprof.enable(gz_fileno, period)
 
     def disable():
         _vmprof.disable()
+        _gzip_finish()
 
     def enable_jitlog(fileno):
         """ Should be a different file than the one provided
             to vmprof.enable(...). Otherwise the profiling data might
             be broken.
         """
-        _vmprof.enable_jitlog(fileno)
+        gz_fileno = _gzip(fileno)
+        _vmprof.enable_jitlog(gz_fileno)
+
+
+_gzip_procs = []
+
+def _gzip(fileno):
+    """Spawn a gzip subprocess that writes compressed profile data to `fileno`.
+
+    Return the subprocess' input fileno.
+    """
+    # Prefer system gzip and fall back to Python's gzip module
+    if which("gzip"):
+        gzip_cmd = ["gzip", "-", "-4"]
+    else:
+        gzip_cmd = ["python", "-m", "gzip"]
+    proc = subprocess.Popen(gzip_cmd, stdin=subprocess.PIPE,
+                            stdout=fileno, bufsize=-1, close_fds=True)
```

Comment:
Thanks. To be honest, I simply copied that from your code when I could not figure out how to properly set up the child process with Python (it never shut down, probably due to the fds being still open). I have given you push access to my fork. If you have the time you can fix it yourself as I won't be getting to it for a few days.